src/arrayMethodLastIndexOf.js

```diff
@@ -5,8 +5,31 @@
  *
  */
 function applyCustomLastIndexOf() {
-  [].__proto__.lastIndexOf2 = function() {
-    // write code here
+  [].__proto__.lastIndexOf2 = function(searchElement, fromIndex = this.length) {
+    if (searchElement === undefined) {
+      return -1;
+    }
+    let index = fromIndex;
+    if (fromIndex < 0) {
+      if (fromIndex < (-this.length)) {
+        return -1;
+      }
+      index = this.length + fromIndex;
+    }
+    for (index; index >= 0; index--) {
+      if (typeof searchElement === typeof this[index]) {
+        if (searchElement === this[index]) {
+          return index;
+        } else if (isNaN(searchElement) && isNaN(this[index])) {
+          return index;
+        } else {
+          continue;
```

Comment:
do not use `continue`, it is bad practice, because hard to maintain `continue`, `break` with nested loops. Moreover, you do not need it at the end of the loop